libraries/AP_NavEKF3/AP_NavEKF3_VehicleStatus.cpp

```diff
@@ -289,6 +289,23 @@ void NavEKF3_core::calcGpsGoodForFlight(void)
     gpsAccuracyGood = gpsSpdAccPass && ekfInnovationsPass;
 }
 
+// Determine if Mag yaw data is good or not
+void NavEKF3_core::calcCompassGoodForFlight()
+{
+    Vector3F eulerAngles;
+    stateStruct.quat.to_euler(eulerAngles.x, eulerAngles.y, eulerAngles.z);
+    ftype velYaw = atan2F(stateStruct.velocity.y,stateStruct.velocity.x);
+
+    // calculate course yaw angle from GPS velocity
+    ftype gpsYaw = atan2F(gpsDataDelayed.vel.y,gpsDataDelayed.vel.x);
```

Comment:
i don't see how this actually works, and it certainly will be wrong for small velocities. When the length of the velocity vector is small then the ground course is essentially random
even with significant velocity we don't really say that it is caused by the compass
I think a far better method would be checking if the compass xyz elements are consistent with the expected earth field

Reply:
note that if the GSF yaw has converged then we could use that yaw to compare

Reply:
@tridge , Shouldn't we use mag variances.